### PR TITLE
[MM-17215] Show different channel not found message for guest

### PR DESCRIPTION
**Pull request by mgdelacroix**
#### Summary
This PR replaces the text that the channel not found page shows for a guest user not to refer to a Town Square channel that they could not be part of.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-17215

Image Pasted at 2019-7-24 08-52

### CODE REVIEW

File: components/error_page/error_page.jsx [hunk 1]
```diff
@@ -73,6 +75,15 @@ export default class ErrorPage extends React.PureComponent {
                     />
                 </Link>
             );
+        } else if (type === ErrorPageTypes.CHANNEL_NOT_FOUND && isGuest) {
+            backButton = (
+                <Link to={params.get('returnTo')}>
```
Comment (jespino) [inline components/error_page/error_page.jsx]: Probably we want to fix that in the place where the `returnTo` is generated, but maybe we want to use `/` directly here

Comment (mgdelacroix) [reply components/error_page/error_page.jsx]: In my testing, the `returnTo` always pointed correctly to the default channel, that's why I didn't consider changing this, but if you have a strong preference for pointing to `"/"`, I'm happy with the change

**Comment by grundleborg** [comment]
@mgdelacroix `i18ncheck` doesn't like it though.

File: components/error_page/error_page.jsx (hunk 1 above)
Comment (jespino) [reply components/error_page/error_page.jsx]: Then I prefer to not change it for now to address the root cause is something goes wrong.

**Comment by srkgupta** [comment]
The changes looks good, however it cannot be tested unless it's pushed to the master instance. Once it's tested, the QA Review label will be removed.

**Pull request merged**